include/Util/NamedPipeListener.h

```diff
@@ -176,8 +175,8 @@ void NamedPipeListener<T>::ListenerThread(const std::function<void(T*)>& callbac
     }
 
     if (listenerData.state == NamedPipeListenerState::Reading) {
-      BOOL fSuccess = ReadFile(listenerData.hPipeInst, listenerData.chRequest, sizeof(T), &listenerData.dwBytesRead, &listenerData.oOverlap);
-      if (fSuccess) {
+      if (const BOOL fSuccess =
+              ReadFile(listenerData.hPipeInst, listenerData.chRequest, sizeof(T), &listenerData.dwBytesRead, &listenerData.oOverlap)) {
```

Comment:
Likewise, I think this can just be `if(ReadFile(...))`

Reply:
Yep

Reply:
These inlining ifs are auto-applied Resharper hints, evidently it wasnt smart enough to see how silly the remaining logic looked.

Reply:
haha, no problem